includes/class-wpcom-legacy-redirector.php

```diff
@@ -81,9 +81,16 @@ static function maybe_do_redirect() {
 		if ( $request_path ) {
 			$redirect_uri = self::get_redirect_uri( $request_path );
 			if ( $redirect_uri ) {
-				header( 'X-legacy-redirect: HIT' );
 				$redirect_status = apply_filters( 'wpcom_legacy_redirector_redirect_status', 301, $url );
-				wp_safe_redirect( $redirect_uri, $redirect_status );
+
+				// Third argument introduced to support the x_redirect_by header to denote WP redirect source.
+				if ( version_compare( get_bloginfo( 'version' ), '5.1.0', '>=' ) ) {
+					wp_safe_redirect( $redirect_uri, $redirect_status, self::POST_TYPE );
```

Comment:
This should be the plugin name, not the name of the CPT it adds IMO.

Reply:
Are you okay with me creating a new const? @GaryJones

Reply:
A new constant in the main plugin file would be fine.